src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,174 @@
+            } else if (word.equals("/by")) {
+                isDescription = false;
+            } else if (isDescription) {
+                description = description.concat(" " + word);
+            } else {
+                by = by.concat(" " + word);
+            }
+        }
+        description = description.trim();
+        by = by.trim();
+        if (description.isBlank()) {
+            throw new DukeException("Oops! The task description cannot be empty.");
+        }
+        if (by.isBlank()) {
+            throw new DukeException("Oops! The task deadline cannot be empty.");
+        }
+        Task task = new Deadline(description, by);
+        TASKS.add(task);
+        System.out.println("     Got it. I've added this task: ");
+        System.out.printf("       %s\n", task);
+        System.out.printf("     Now you have %d tasks in the list.\n", TASKS.size());
+    }
+
+    private static void handleEvent(String line) throws DukeException {
+        String[] input = line.split(" ");
+        String description = "";
+        String at = "";
+        boolean isDescription = true;
+        for (String word: input) {
+            if (word.equals("event")) {
+                continue;
+            } else if (word.equals("/at")) {
+                isDescription = false;
+            } else if (isDescription) {
+                description = description.concat(" " + word);
+            } else {
+                at = at.concat(" " + word);
+            }
+        }
+        description = description.trim();
+        at = at.trim();
+        if (description.isBlank()) {
+            throw new DukeException("Oops! The task description cannot be empty.");
+        }
+        if (at.isBlank()) {
+            throw new DukeException("Oops! The task dates/times cannot be empty.");
+        }
+        Task task = new Event(description, at);
+        TASKS.add(task);
+        System.out.println("     Got it. I've added this task: ");
+        System.out.printf("       %s\n", task);
+        System.out.printf("     Now you have %d tasks in the list.\n", TASKS.size());
+    }
+
+    private static void handleException() {
+        System.out.println("     Oops! I'm sorry. I don't know what that means.");
+    }
+
+    /**
+     * Main method of duke project.
+     */
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+
+        String greeting = "     Hello! I'm Duke\n"
+                + "     What can I do for you?";
+        System.out.println(greeting);
+
+        while (sc.hasNext()) {
+            String line = sc.nextLine();
+            if (line.equals("bye")) {
+                System.out.println("     Bye. Hope to see you again soon!");
+                break;
+            }
+            if (line.equals("list")) {
+                handleList();
+                continue;
+            }
+            String[] input = line.split(" ");
+            try {
+                switch (input[0]) {
```

Comment:
Overall flow of the main code is logical and correct, nice. Also, if you would like, you can use input[0].toLowerCase() to for greater leeway for the user, unless you want it to be more strict.